### ISSUE TITLE
2 of 3 unit tests fail in `test_pydantic_polymorphism.py`

### ISSUE THREAD

**Issue by capsulecorplab** [linkml/linkml#1538]
**Describe the bug**
`test_pydantic_load_poly_data` and `test_type_hierarchy` fail when running unit tests for [test_pydantic_polymorphism.py](https://github.com/linkml/linkml/blob/v1.5.6/tests/test_issues/test_pydantic_polymorphism.py) as of linkml [v1.5.6](https://github.com/linkml/linkml/tree/v1.5.6)

**To reproduce**
1. `pip install linkml==1.5.6`
2. clone and `cd` into linkml repo [v1.5.6](https://github.com/linkml/linkml/tree/v1.5.6)
3. run `python -m unittest tests/test_issues/test_pydantic_polymorphism.py `

**Expected behavior**
Should pass unit tests

**Screenshots**
![Screenshot from 2023-07-14 15-34-23](https://github.com/linkml/linkml/assets/14095576/cce2bd30-490a-4a9a-ac5d-20341f97462a)

**About your computer (if applicable, please complete the following information):**
Ubuntu Focal (20.04)

**Additional context**
Follows #1224 #1275 #1431

**Comment by sierra-moxon** [comment]
Hi @capsulecorplab - could you help me understand your workflow here?  I can't seem to replicate it within my virtual environment.   The idea here is that you install linkml globally outside of a virtual environment?

**Comment by capsulecorplab** [comment]
> Hi @capsulecorplab - could you help me understand your workflow here? I can't seem to replicate it within my virtual environment. The idea here is that you install linkml globally outside of a virtual environment?

Correct, I'm running a dockerized virtual environment that pip installed linkml v1.5.6 globally as part of the image build https://github.com/Mach30/kasm-dof-workspace
